src/generators/types.ts

```diff
@@ -0,0 +1,45 @@
+import { Entity } from "../types";
+import { ApiEntitySpec } from "./kinds/api";
+import { ComponentEntitySpec } from "./kinds/component";
+import { DomainEntitySpec } from "./kinds/domain";
+import { GroupEntitySpec } from "./kinds/group";
+import { LocationEntitySpec } from "./kinds/location";
+import { ResourceEntitySpec } from "./kinds/resource";
+import { SystemEntitySpec } from "./kinds/system";
+import { TemplateEntitySpec } from "./kinds/template";
+import { UserEntitySpec } from "./kinds/user";
+
+export interface EntityProfile {
+  displayName?: string;
+  email?: string;
+  picture?: string;
+}
+
+export type WellKnownLifecycle = 'experimental' | 'production' | 'deprecated';
+
+export interface BaseEntitySpec {
+  type?: string;
+  owner: string;
```

Comment:
_Bit of a pain_ but I think we need to hide the (or disable) the owner field in the form if the kind is one of user, group, location 👍

Reply:
We now override the owner field in a group or user spec, along with hiding the owner text field when a group or user entity is selected to be generated.